feat(views): mostrar calendario por areas, reservar y cancelar reserva de areas comunales.

Cristian-Zambrano opened the pull request:
feat (views): mostrar calendario por areas, reservar y cancelar reserva de areas comunales.

cristian-sangucho-a committed:
test: escritura de caracteristica

Cristian-Zambrano committed:
test: crear escenario reserva de espacio disponible para actividad publica

Cristian-Zambrano committed:
test: crear escenario reserva de un espacio publico para actividad privada

Cristian-Zambrano committed:
test: crear escenario 'Cancelar reserva de actividad publica'

Cristian-Zambrano committed:
test: crear escenario 'Cancelar reserva de actividad privada'

cristian-sangucho-a committed:
test: actualizar caracteristicas

cristian-sangucho-a committed:
test: crear los pasos de todos los escenarios

Cristian-Zambrano committed:
test: corrección grupal del feature - versión 1.0

Cristian-Zambrano committed:
test: implementar los steps para los escenarios reservar area comunar para actividad privada y publica del feature en la versión 1.0

JorMath committed:
test: realizar actividad grupal: Cancelar reserva de actividad publica en area comunal, cambio en el .feature

JorMath committed:
test: realizar actividad grupal: Cancelar reserva de actividad publica en area comunal, puesto el step para cancele la reserva

jeanpcot committed:
test: Actualizar escenarios Cancelar reserva de actividad publica/privada en area comunal

jeanpcot committed:
test: Realizar pasos correspondientes a Cancelar actividad publica/privada en area comunal

jeanpcot committed:
Merge branch 'develop' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
test: realizar las entidades en los test utilizando los modelos definidos en ciudadano_app/models

Cristian-Zambrano committed:
test: correccion de un paso en feature.

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/develop' into feature/realizar_actividades_grupales

# Conflicts:
#	features/realizar_actividades_grupales.feature

JorMath committed:
feat(models): agregar entidades area_comunal, espacio_publico

JorMath committed:
test: actualizar entidad que controla los espacios públicos

JorMath committed:
tests: cambio de creación de modelos para el primer step

cristian-sangucho-a committed:
feat: verificar la reservas activas del ciudadano

JorMath committed:
fix: configuracion de merge

JorMath committed:
feat(model) cambio en servicio_reserva, uso de mediator

cristian-sangucho-a committed:
refactor(models): quitar la negacion de la declaracion

Cristian-Zambrano committed:
feat: aagregar parametros de la clase reserva y creacion de metodos para agregar correos invitados a la reserva.

cristian-sangucho-a committed:
feat(servicio_reserva): reservar una area comunal

Cristian-Zambrano committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/models/reserva.py

cristian-sangucho-a committed:
fix(steps): corregir la negacion

cristian-sangucho-a committed:
fix(steps): corregir la negacion

Cristian-Zambrano committed:
test: corregir el parametro de la funcion enviar_notificacion

Cristian-Zambrano committed:
test: agregar assert para el agregado de correo

Cristian-Zambrano committed:
feat (reserva): agregar metodos para obtencion de los correos y del area comunal de la reserva.

Cristian-Zambrano committed:
feat (reserva): implementar funcion para el envio de la notificacion de una invitacion privada.

JorMath committed:
fix (models): re-escribir el init.py para que lea correctamente las importaciones relativas

JorMath committed:
Merge branch 'feature/realizar_actividades_grupales' of https://github.com/juanfcarrillo/servicios-ciudadanos into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/models/reserva.py

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/models/reserva.py

JorMath committed:
fix (models): importar en un orden específico para no romper las relaciones con la bd 🤮

JorMath committed:
Merge branch 'feature/realizar_actividades_grupales' of https://github.com/juanfcarrillo/servicios-ciudadanos into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

JorMath committed:
fix (models): cambiar la herencia de la super clase Model de django en los servicios, y ciertos campos para migrar correctamente

JorMath committed:
Merge branch 'feature/realizar_actividades_grupales' of https://github.com/juanfcarrillo/servicios-ciudadanos into feature/realizar_actividades_grupales

JorMath committed:
fix (model): migrar para que acepte valores null hasta llenar la bd con datos, y corrección a foreign key de reserva hacia area_comunal

Cristian-Zambrano committed:
feat (entidades): agregar campo para mapeo del espacio publico en el area comunal

Cristian-Zambrano committed:
test: actualizar del paso dado en el primer y segundo escenario

Cristian-Zambrano committed:
test: agregar interfaz para implementar los metodos de prueba en respositorio_reserva_en_memoria

Cristian-Zambrano committed:
amend: borrar metodo no utilizado

Cristian-Zambrano committed:
test: actualizar primer paso de los escenarios

JorMath committed:
test: aprobar que no supera el ciudadano las 3 reservas

Cristian-Zambrano committed:
test: actualizar metodos necesarios para pasar el primer escenario

Cristian-Zambrano committed:
test: implementar pasos para pasar test

Cristian-Zambrano committed:
test: implementar funciones en repositorio_reserva_en_memoria.py

Cristian-Zambrano committed:
test: corregir step

Cristian-Zambrano committed:
feat (modelos): agregar nombre de relacion para el Espacio publico y la entidad municipal

Cristian-Zambrano committed:
amend: correccion de variable

Cristian-Zambrano committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
fix: merge con develop

cristian-sangucho-a committed:
test:  agregar los correo a los invitados

jeanpcot committed:
test(Escenario 1): Implementar los dos ultimos steps

jeanpcot committed:
fix(migrations): entidad municipal

jeanpcot committed:
fix(migrations): entidad municipal

jeanpcot committed:
fix(step): Quitar random al Id del area comunal para obtenerla por medio de este

jeanpcot committed:
fix(funcion): Usar el Id del area comunal para obtener una especifica

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
test: completar test del primer, segundo y parte del tercer escenario

jeanpcot committed:
fix(escenario): Reservar de manera privada, faltaba un paso necesario

jeanpcot committed:
fix(step): Reservar de manera privada, no se verificaba si existian correos

jeanpcot committed:
fix(step): Reservar de manera privada, el metodo de enviar invitaciones utilizaba el modelo.

jeanpcot committed:
fix(servicio): Reservar de manera privada, se estaba usando el modelo, ahora se usa una simulacion de este

jeanpcot committed:
test(escenario): Cancelar reserva privada/publica

jeanpcot committed:
add(model): Interfaz de servicio de reservas

jeanpcot committed:
Merge branch 'develop' into feature/realizar_actividades_grupales

jeanpcot committed:
fix(import): Reporte y Tipo Reporte

jeanpcot committed:
fix(migrations)

jeanpcot committed:
fix(tests): Quitar interfaz

jeanpcot committed:
fix(code): Revision del SonarQube

jeanpcot committed:
fix(code): Revision del SonarQube

jeanpcot committed:
update(packages): dotenv

jeanpcot committed:
fix(feature): Borrar los TODO

jeanpcot committed:
fix(steps): Colocar asserts

cristian-sangucho-a committed:
refactor(models): agrupar las clases de reseva

cristian-sangucho-a committed:
refactor(step): extraer metodo para crear la reserva

cristian-sangucho-a committed:
refactor(step): eliminar comentarios innecesarios

cristian-sangucho-a committed:
refactor: resolver asuntos reportados por sonarqube

JorMath committed:
feat (model): agregar principales overrides que se realizan con el ORM de django

JorMath committed:
feat (model): agregar principales overrides que se realizan con el ORM de django

JorMath committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/models/reserva/servicio_reserva.py

JorMath committed:
Merge branch 'develop' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
feat (envio de notificaciones por correo): implementar metodos en la clase ServicioNotificacionPorCorreo y configuraciones en settings para el envio de correos con django core mail

Cristian-Zambrano committed:
feat (envio de notificaciones por correo): implementar metodos en la clase ServicioNotificacionPorCorreo y configuraciones en settings para el envio de correos con django core mail

Cristian-Zambrano committed:
Merge branch 'feature/realizar_actividades_grupales' of https://github.com/cristian-sangucho-a/servicios-ciudadanos into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
feat(models): reservar area comunal para actividad privada

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
feat(models): cancelar reserva de actividad privada

cristian-sangucho-a committed:
feat(models): cancelar reserva de actividad privada

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
chore (dependencias): instalar tailwind

Cristian-Zambrano committed:
fix (merge): arreglar conflictos

Cristian-Zambrano committed:
fix (merge): arreglar conflictos

Cristian-Zambrano committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
docs: agregar dependencia de node

jeanpcot committed:
fix(settings): NPM_BIN_PATH

jeanpcot committed:
fix(views): Obtener ciudadano y mandar a dashboard luego de iniciar sesion

cristian-sangucho-a committed:
feat(view): crear formulario de reserva

Cristian-Zambrano committed:
feat (calendario): presentar calendario en agenda publica

cristian-sangucho-a committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/urls.py
#	ciudadano_app/views/dashboard.py
#	ciudadano_app/views/login.py
#	entidad_municipal_app/admin.py
#	entidad_municipal_app/models/espacio_publico.py

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/urls.py
#	entidad_municipal_app/admin.py

jeanpcot committed:
fix(merge): Resolver conflictos con develop

jeanpcot committed:
fix(conflictos): merge

cristian-sangucho-a committed:
fix(views): mostrar el input de los correos si la reserva es privada

cristian-sangucho-a committed:
feat(views): mostrar la informacion del ciudadano y el area comunal a reservar

cristian-sangucho-a committed:
style(views): hacer reponsive el formulario de reserva

Cristian-Zambrano committed:
feat (calendario-reservas): mostrar reservas por dia seleccionado en el calendario

Cristian-Zambrano committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
Merge branch 'feature/realizar_actividades_grupales' of https://github.com/cristian-sangucho-a/servicios-ciudadanos into feature/realizar_actividades_grupales

Cristian-Zambrano committed:
fix (clases): agregar clases faltantes para mis reservas

jeanpcot committed:
feat(views): Cancelar reserva

# Conflicts:
#	ciudadano_app/urls.py
#	ciudadano_app/views/__init__.py

jeanpcot committed:
refactor(models): Cancelar reserva

Cristian-Zambrano committed:
feat: agregar parametros en url para vista de reserva.py

cristian-sangucho-a committed:
feat(views): mostrar las reservas del ciudadano

cristian-sangucho-a committed:
fix(views): redireccionar a mis_reservas cuando se cancela

cristian-sangucho-a committed:
fix(models): reservar para actividades publicas y privadas

cristian-sangucho-a committed:
refactor(views): renombrar la ruta

cristian-sangucho-a committed:
refactor(views): renombrar la ruta

cristian-sangucho-a committed:
refactor(views): renombrar la ruta

cristian-sangucho-a committed:
feat(views): guardar la reserva

cristian-sangucho-a committed:
refactor(models): cambiar la firma del metodo para recibir los correos de os invitados

cristian-sangucho-a committed:
refactor(models): cambiar la firma del metodo para recibir los correos de os invitados

Cristian-Zambrano committed:
refactor(views): limpiar codigo en reserva

Cristian-Zambrano committed:
fix (model): arreglar el metodo obtener_reservas_area_comunal para reservas activas

cristian-sangucho-a committed:
feat(templates):  agregar boton para ir a la agenda

cristian-sangucho-a committed:
feat(templates):  agregar boton para ir a la agenda

Cristian-Zambrano committed:
fix (model): crear metodo para validar que no se sobreescriba una reserva

Cristian-Zambrano committed:
refactor (model): refactor de metodo repetido

# Conflicts:
#	ciudadano_app/models/reserva/servicio_reserva.py
#	ciudadano_app/views/reserva.py

Cristian-Zambrano committed:
chore (root): eliminar .env

Cristian-Zambrano commented:
CDN HTMX: se refactorizará mañana a las 3 p.m. por el htmx de Django.

juanfcarrillo reviewed (commented):
Muchachos, por que usan htmx ?

cristian-sangucho-a committed:
chore(static): agregar dependencia de htmx para consultas dinamicas

juanfcarrillo commented on ciudadano_app/templates/agenda.html:
```diff
@@ -0,0 +1,44 @@
+                            class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
+                            hx-get="{% url 'cargar_areas_comunales' %}"
+                            hx-trigger="change"
+                            hx-target="#areas-container"
```
Este select se puede hacer con el sistema de templates de django. Revisar enviar_reporte.py y enviar_reporte.html

cristian-sangucho-a replied:
se cambio a js

juanfcarrillo commented on ciudadano_app/templates/areas_comunales.html:
```diff
@@ -0,0 +1,14 @@
+    class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline"
+    hx-get="{% url 'cargar_calendario' %}"
+    hx-trigger="change"
+    hx-target="#calendario-container"
```
Misma observacion

juanfcarrillo commented:
Creo que no es lo mejor agregar una dependencia como HTMX para 2 templates. Eso evita la consistencia y mantenibilidad.

cristian-sangucho-a committed:
style(forms): cambiar a text area el input de los correos

Cristian-Zambrano committed:
fix(templates): arreglar eleccion de dias pasados mostradas en calendario para reserva.

Cristian-Zambrano committed:
fix(views): correccion aplicada del informe del sonarqube

JorMath committed:
chore (environ): usar la librería de environ de django en vez de la del sistema operativo

JorMath committed:
chore (fix): agregar dependencia de htmx al requirements

JorMath committed:
docs(steps): documentar los steps

JorMath committed:
docs(model): documentar para los servicios de reserva y notificación

JorMath committed:
docs(model): documentar para el modelo de reserva y area comunal

JorMath committed:
docs(views): documentar las views.py de la reserva

Cristian-Zambrano committed:
refactor(views): cambiar HTMX por JavaScript y solucionar problema de nombramiento en variable en servicio_reserva.py

# Conflicts:
#	ciudadano_app/models/reserva/servicio_reserva.py

# Conflicts:
#	ciudadano_app/templates/agenda.html

cristian-sangucho-a committed:
Revert "chore(static): agregar dependencia de htmx para consultas dinamicas"

This reverts commit e1a4d121f4f5f7a11b285ed5622f857458a43bd1.

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/develop' into feature/realizar_actividades_grupales

# Conflicts:
#	ciudadano_app/templates/dashboard.html
#	ciudadano_app/urls.py
#	ciudadano_app/views/__init__.py
#	entidad_municipal_app/models/espacio_publico.py
#	requirements.txt
#	servicios_ciudadanos/settings.py

cristian-sangucho-a committed:
fix(general): resolver los asuntos de sonarqube

cristian-sangucho-a committed:
fix(general): resolver los asuntos de sonarqube

JorMath committed:
fix(ci): update a ci.yml para las vars .env

JorMath committed:
fix(ci): update a ci para .env

riofutabac commented:
perdoname ñaño recien hice el de ciudadano

cristian-sangucho-a committed:
fix(general): resolver los asuntos de sonarqube

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

juanfcarrillo committed:
Merge branch 'develop' into feature/realizar_actividades_grupales

cristian-sangucho-a committed:
style(templates): agregar estilo para ver la agenda publica

cristian-sangucho-a committed:
Merge remote-tracking branch 'origin/feature/realizar_actividades_grupales' into feature/realizar_actividades_grupales

juanfcarrillo reviewed (approved):
Buen trabajo muchachos 🚀

cristian-sangucho-a committed:
fix(templates): resolver merge

juanfcarrillo reviewed (approved):
Esta es la buena
